store/src/java/com/zimbra/cs/service/admin/CreateVolume.java

```diff
@@ -50,13 +48,22 @@ private CreateVolumeResponse handle(CreateVolumeRequest request, Map<String, Obj
         checkRight(zsc, ctx, Provisioning.getInstance().getLocalServer(), Admin.R_manageVolume);
 
         VolumeInfo volInfoRequest = request.getVolumeInfo();
+
         Volume.StoreType enumStoreType = (1 == volInfoRequest.getStoreType()) ? Volume.StoreType.INTERNAL : Volume.StoreType.EXTERNAL;
+
         VolumeConfigUtil.validateCreateVolumeRequest(request, volInfoRequest, enumStoreType);
 
         Volume volRequest = VolumeManager.getInstance().create(toVolume(volInfoRequest, enumStoreType));
         VolumeInfo volInfoResponse = volRequest.toJAXB();
         VolumeConfigUtil.postCreateVolumeActions(request, volRequest, volInfoRequest, volInfoResponse, enumStoreType);
-        return new CreateVolumeResponse(volInfoResponse);
+
+        CreateVolumeResponse createVolumeResponse = new CreateVolumeResponse(volInfoResponse);
+        // if its primary volume then set respective store manager if it's different
+        if (volInfoRequest.isCurrent() && volInfoRequest.getType() == Volume.TYPE_MESSAGE) {
+            StoreManagerRuntimeSwitchResult runtimeSwitchResult = StoreManagerResetHelper.setNewStoreManager(volInfoRequest.getStoreManagerClass());
```

Comment:
no further use of var `runtimeSwitchResult` can remove it and  directly use `StoreManagerResetHelper.setNewStoreManager(volInfoRequest.getStoreManagerClass())` in the call

Reply:
This helps us in debugging.